code/modules/mob/living/living_defense.dm

```diff
@@ -19,6 +19,11 @@
 	if(prob(armor * modifier))
 		absorb += 1
 
+	if(prob(armor_penetration))
+		absorb -= 1
+	if(prob(armor_penetration))
+		absorb -= 1
+
```

Comment:
This seems like a really shitty way of doing this.

Reply:
What's not to like?

absorb += 1s if armor succesfully rolls on probability, so absorb -= 1s on prob(armor_penetration)

If absorb >= 2, it returns 2, if absorb == 1 it returns 1, else it returns 0, so it's safe is absorb is below 0